sqlx-aws/examples/dsql.rs

```diff
@@ -0,0 +1,17 @@
+use aws_sdk_dsql::error::BoxError;
+use sqlx_aws::iam::dsql::DsqlIamProvider;
+use sqlx_postgres::{PgConnectOptions, PgPoolOptions};
+
+#[tokio::main]
+async fn main() -> Result<(), BoxError> {
+    let hostname = std::env::var("DSQL_CLUSTER_ENDPOINT")
+        .expect("please set DSQL_CLUSTER_ENDPOINT is your environment");
```

Comment:
For reviewers: I didn't want to write an automated test for this, because the test would fail unless the developer had AWS credentials. This seems like a good compromise for now. Would be interested to know if there's another viable approach.